src/challenge/lock-logic.js

```diff
@@ -15,16 +19,33 @@ function changeDialValue (index, incrementBy) {
   // this will be called with arguments (0, 1) indicating we should raise the first dial's value by one
   // for example, if the user clicked the "down" arrow for the last wheel
   // this will be called with arguments (3, -1).
-
   // to change the state of the lock, simply make a call like
   // lockState.locked = false
   // or lockState.wheels[1] = 2
   // the lock will re-render itself when the value changes
+  const currentValue = lockState.wheels[index]
+  const newValue = currentValue + incrementBy
+  lockState.wheels[index] = newValue
 
   // When the lock is set to match the secretCombo
   // call the redirect() function with your name
   // eg: redirect('larry-lobster')
   // the redirect function will only redirect if the lockState is unlocked
+  _checkIfAllLocksAreMatch()
+}
+
+function _checkEqualLockNumber (combination) {
+  const shouldUnlocked = lockState.wheels[0] === SECRET_COMBO[0] && lockState.wheels[1] === SECRET_COMBO[1] && lockState.wheels[2] === SECRET_COMBO[2] && lockState.wheels[3] === SECRET_COMBO[3]
```

Comment:
please de-couple the length of the lock from the verification of the combination

Reply:
I am not sure what de-couple the length of the lock

Do you mean extract method? or do you mean formatted code?

Reply:
I have improved the way to check lock's combination.